plugins/woocommerce/src/Internal/RestApi/Routes/V4/ShippingZones/Controller.php

```diff
@@ -230,6 +242,38 @@ public function create_item( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Delete a shipping zone by zone id.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object or WP_Error.
+	 */
+	public function delete_item( $request ) {
+		$zone_id = (int) $request['id'];
+		$force   = $request['force'];
+
+		// Shipping zones do not support trashing.
+		// This is done to ensure backward compatibility w/ v2/v3.
+		if ( ! $force ) {
```

Comment:
I believe it's fine to remove the need for this in the request before delete can happen, if the caller wants to delete, they want to delete.
We should however clearly mention this in the docs.